config/routes.rb

```diff
@@ -5,10 +5,10 @@
   get '/pages/guidance', to: redirect('/about-bt-wifi')
 
   resources :application_forms do
-    get 'success/:application_form_id', to: 'application_forms#success', as: :success
+    get 'success/:recipient_id', on: :collection, to: 'application_forms#success', as: :success
   end
   resources :allocation_request_forms do
-    get 'success', to: 'allocation_request_forms#success'
+    get 'success/:allocation_request_id', on: :collection, to: 'allocation_request_forms#success', as: :success
```

Comment:
This can be simplified as:

```suggestion
    get :success
```

Which would generate the route:

```
allocation_request_form_success GET    /allocation_request_forms/:allocation_request_form_id/success(.:format)      allocation_request_forms#success
```